pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -209,6 +215,13 @@ func (cb *ClusterBuilder) applyDestinationRule(mc *MutableCluster, clusterMode C
 	return subsetClusters
 }
 
+func (cb *ClusterBuilder) applyMetadataExchange(c *cluster.Cluster) {
+	spec := cb.push.Telemetry.EffectiveTelemetry(cb.proxy)
```

Comment:
EnvoyFilter based solution applies TCP MX exchange filter regardless if the proxy has metrics enabled or not. I think the upstream peer should be able to get source metadata regardless of whether client has metric enabled or not? Otherwise upstream stats will have missing labels.

Reply:
I see, we can switch to AnyTelemetryExists then?

Reply:
Yeah I think so.